lib/Infer/ExhaustiveSynthesis.cpp

```diff
@@ -166,6 +166,9 @@ void getGuesses(std::vector<Inst *> &Guesses,
     }
   }
 
+  Inst *I1 = IC.getReservedInst();
```

Comment:
This reverts some wrong changes made recently. A case failed because of that change.